XenAdmin/Wizards/ImportWizard/ImportSelectNetworkPage.cs

```diff
@@ -51,6 +54,31 @@ protected override bool ImplementsIsDirty()
             return true;
         }
 
+        protected override bool ShowNetwork(Host targetHost, XenAPI.Network network, string sysId = null)
+        {
+            if (network.IsSriov() && !AllowSriovNetwork(sysId))
+                return false;
+
+            return base.ShowNetwork(targetHost, network);
+        }
+
+        private bool AllowSriovNetwork(string sysId)
+        {
+            var vhs = OVF.FindVirtualHardwareSectionByAffinity(SelectedOvfEnvelope, sysId, "xen");
+            var data = vhs.VirtualSystemOtherConfigurationData;
+
+            foreach (var s in data)
+            {
+                if (s.Name == "recommendations")
+                {
+                    if (s.Value.Value.Contains("allow-network-sriov"))

```

Comment:
I think we need to read the s.Value.Value as xml and get the actual value of "allow-network-sriov" here, because this will still give the wrong answer if you have `restriction field="allow-network-sriov" value="0"`.
The method VM.HasSriovRecommendation() is not checking for the value either, I think in both cases we should repeat the same logic as in the method VM.CanHaveGpu().

Reply:
For this problem, we only have the "allow-network-sriov" field for thoes VM which allows sriov. For those VMs which doesn't allow sriov, the "allow-network-sriov" doesn't appear. So we can just check the "allow-network-sriov" value.